sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs

```diff
@@ -36,11 +45,38 @@ public class ProcessMessageEventArgs : EventArgs
         /// <param name="cancellationToken">The processor's <see cref="System.Threading.CancellationToken"/> instance which will be cancelled
         /// in the event that <see cref="ServiceBusProcessor.StopProcessingAsync"/> is called.
         /// </param>
-        public ProcessMessageEventArgs(ServiceBusReceivedMessage message, ServiceBusReceiver receiver, CancellationToken cancellationToken)
+        public ProcessMessageEventArgs(ServiceBusReceivedMessage message, ServiceBusReceiver receiver, CancellationToken cancellationToken) :
+            this(message, manager: null, cancellationToken)
         {
-            Message = message;
             _receiver = receiver;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ProcessMessageEventArgs"/> class.
+        /// </summary>
+        ///
+        /// <param name="message">The message to be processed.</param>
+        /// <param name="manager">The receiver manager for these event args.</param>
+        /// <param name="cancellationToken">The processor's <see cref="System.Threading.CancellationToken"/> instance which will be cancelled
+        /// in the event that <see cref="ServiceBusProcessor.StopProcessingAsync"/> is called.
+        /// </param>
+        internal ProcessMessageEventArgs(
+            ServiceBusReceivedMessage message,
+            ReceiverManager manager,
+            CancellationToken cancellationToken)
+        {
+            Message = message;
+            Messages[message] = default;
+            _manager = manager;
```

Comment:
The manager's unmanaged objects would still get cleaned up. GC doesn't trigger the clean up in any case.

Reply:
> GC doesn't trigger the clean up in any case.

Except in the case of finalizers, which I believe are used in several places in the network areas of the BCL.   That said, I was more thinking about the wasted memory in this case.   I don't have a strong concern.

Reply:
Ah, I see your point - the explicit cleanup that we are doing in the processor will still happen regardless of whether the user is holding on to a reference, so it shouldn't matter whether or not the finalizer runs, at least I don't think it would.